lib/Zod.ts

```diff
@@ -1,21 +1,29 @@
-import { ZodError, ZodIssue, ZodSchema, z } from "zod"
-import { Constructor } from "./Types/HelperTypes"
+import { ZodEffects, ZodError, ZodIssue, ZodSchema, ZodType, z } from "zod"
+import { Prototype } from "./Types/HelperTypes"
 
-const optionalParse = <Input, Output extends Constructor>(
-  constructor: Output,
-  parse: (input: Input) => InstanceType<Output>
+/**
+ * optionalParse creates a Zod schema that accepts either an instance of a class
+ * or raw input that can be transformed into an instance of that class.
+ *
+ * @param clazz - The class (can have public or private constructor).
+ * @param schema - A Zod schema that validates and transforms raw input into the desired Output type.
+ * @returns A Zod schema that accepts either an instance of the class or raw input.
+ */
+const optionalParse = <Input, Output extends Prototype>(
```

Comment:
```suggestion
const optionalParse = <Input, Def extends ZodTypeDef = ZodTypeDef, Output extends Prototype>(
```

Reply:
I got type errors when trying to make it generic. I'll leave it as ZodTypeDef